Hermes/hermes.py

```diff
@@ -117,6 +146,94 @@ def send_all_stats():
+
+  uaserver = SOAPpy.SOAPProxy('https://{0}:{1}'.format(
+    appscale_info.get_db_master_ip(), UA_SERVER_PORT))
+
+  # If the appscalesensor app is already running, then do nothing.
+  is_app_enabled = uaserver.is_app_enabled(APPSCALE_SENSOR,
+    appscale_info.get_secret())
+  if is_app_enabled == "true":
+    logging.debug("{} application is already running.".format(APPSCALE_SENSOR))
+    return
+
+  pwd = encrypt_password(USER_EMAIL, random_password_generator())
+  if create_appscale_user(pwd, uaserver) and create_xmpp_user(pwd, uaserver):
+    logging.info("Created new user and now tarring app to be deployed.")
+    file_path = os.path.join(os.path.dirname(__file__), '../Apps/sensor')
+    file_suffix = 'tar.gz'
+    app_dir_location = os.path.join(hermes_constants.APP_DIR_LOCATION,
+      APPSCALE_SENSOR)
+    archive = tarfile.open(app_dir_location, "w|gz")
+    archive.add(file_path, arcname= APPSCALE_SENSOR)
+    archive.close()
+
+    try:
+      logging.info("Deploying the sensor app for registered deployments.")
+      acc = appscale_info.get_appcontroller_client()
+      acc.upload_app(app_dir_location, file_suffix, USER_EMAIL)
+
+    except AppControllerException:
+      logging.exception("AppControllerException while trying to deploy "
+        "appscalesensor app.")
+  else:
+    logging.error("Error while creating or accessing the user to deploy "
+      "appscalesensor app.")
+
+def create_appscale_user(password, uaserver):
+  """ Creates the user account with the email address and password provided. """
+  does_user_exist = uaserver.does_user_exist(USER_EMAIL, appscale_info.get_secret())
+  if does_user_exist == "true":
+    logging.info("User {0} already exists, so not creating it again.".
+      format(USER_EMAIL))
+    return True
+  else:
+    if uaserver.commit_new_user(
+      USER_EMAIL, password, ACCOUNT_TYPE, appscale_info.get_secret()) == "true":
+      return True
+    else:
+      logging.error("Error while creating an Appscale user.")
+      return False
+
+def create_xmpp_user(password, uaserver):
+  """ Creates the XMPP account. If the user's email is a@a.com, then that
+  means their XMPP account name is a@login_ip. """
+  username_regex = re.compile('\A(.*)@')
+  username = username_regex.match(USER_EMAIL).groups()[0]
+  xmpp_user = "{0}@{1}".format(username, appscale_info.get_login_ip())
+  xmpp_pass = encrypt_password(xmpp_user, password)
+  does_user_exist = uaserver.does_user_exist(xmpp_user,
+    appscale_info.get_secret())
+  if does_user_exist == "true":
+    logging.info("XMPP User {0} already exists, so not creating it again.".
+      format(xmpp_user))
+    return True
+  else:
+    if uaserver.commit_new_user(
+      xmpp_user, xmpp_pass, ACCOUNT_TYPE, appscale_info.get_secret()) == "true":
+      logging.info("XMPP username is {0}".format(xmpp_user))
+      return True
+    else:
+      logging.error("Error while creating an XMPP user.")
+      return False
+
+def random_password_generator():
```

Comment:
This should be moved somewhere else. Probably in appscale/lib. If there's not utils file you can make one.